python/tank/platform/qt/pyside2_patcher.py

```diff
@@ -0,0 +1,233 @@
+        """
+        @staticmethod
+        def setCodecForCStrings(codec):
+            # Empty stub, doesn't exist in Qt5.
+            pass
+
+        QTextCodec.setCodecForCStrings = setCodecForCStrings
+
+    @classmethod
+    def _patch_QCoreApplication(cls, QCoreApplication):
+        """
+        Patches QCoreApplication.
+
+        :param QCoreApplication: The QCoreApplication class.
+        """
+        translate = QCoreApplication.translate
+
+        @staticmethod
+        def wrapper(context, source_text, disambiguation=None, encoding=None, n=None):
+            # In PySide2, the encoding argument has been deprecated, so we don't
+            # pass it down. n is still supported however, but has always been optional
+            # in PySide. So if n has been set to something, let's pass it down,
+            # otherwise Qt5 has a default value for it, so we'll use that instead.
+            if n is not None:
+                return translate(context, source_text, disambiguation, n)
+            else:
+                return translate(context, source_text, disambiguation)
+
+        QCoreApplication.translate = wrapper
+
+        QCoreApplication.CodecForTr = 0
+        QCoreApplication.UnicodeUTF8 = 1
+        QCoreApplication.DefaultCodec = QCoreApplication.CodecForTr
+
+    @classmethod
+    def _patch_QApplication(cls, QtGui):
+        """
+        Patches QApplication.
+
+        :param QtGui: QtGui module.
+        """
+        QApplication = QtGui.QApplication
+
+        class Wrapper(QApplication):
+            def __init__(self, *args):
+                QApplication.__init__(self, *args)
+                # qApp has been moved from QtGui to QtWidgets, make sure that
+                # when QApplication is instantiated than qApp is set on QtGui.
+                QtGui.qApp = self
+
+            @staticmethod
+            def palette():
+                # Retrieve the application palette by passing no widget.
+                return QApplication.palette(None)
+
+        QtGui.QApplication = Wrapper
+
+    @classmethod
+    def _patch_QAbstractItemView(cls, QtGui):
+        """
+        Patches QAbstractItemView.
+
+        :param QtGui: QtGui module.
+        """
+        QAbstractItemView = QtGui.QAbstractItemView
+
+        class Wrapper(QAbstractItemView):
+            def __init__(self, *args):
+                QAbstractItemView.__init__(self, *args)
+                # dataChanged's signature has an extra parameter in Qt5. Since this
+                # is a virtual method than is usually overriden, monkey patch
+                # this object's dataChanged if present to Qt can invoke the Python
+                # version and then forward the call back to the derived class's
+                # implementation.
+                if hasattr(self, "dataChanged"):
+                    dataChanged = self.dataChanged
+                    self.dataChanged = lambda tl, br, roles: dataChanged(tl, br)
```

Comment:
An inline method definition would be much easier to read (and comment) than the lambda.

Also, is the comment right?  This implementation means that the extra arg is in Qt4, not Qt5, right?  I might be backward but that seems to be what the patch is doing.